src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -5366,7 +5366,14 @@ private bool TryReadSqlStringValue(SqlBuffer value, byte type, int length, Encod
                             }
                             if (length > 0)
                             {
-                                s = new string(cc, 0, length);
+                                if (cc.LongLength == length + 1)
```

Comment:
This is the actual fix isn't it? Can you explain the problem that was found?

Reply:
I'm still investigating hence WIP, wanted to try out tests in various environments in order to validate changes. I'll share the exact solution likely to be once I get something working.
Thanks for chiming in!